docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,296 @@
+
+    ```go
+        const serviceKey = "app-myservice"
+        ...
+    
+        service, ok := pkg.NewAppService(serviceKey)
+        if !ok {
+            os.Exit(-1)
+        }
+    ```
+
+!!! example "Example - Create Application Service instance with Target Type specified"
+
+    ```go
+        const serviceKey = "app-myservice"
+        ...
+    
+        service, ok := pkg.NewAppServiceWithTargetType(serviceKey, &[]byte{})
+        if !ok {
+            os.Exit(-1)
+        }
+    ```
+
+Since the factory function logs all errors all you need to do is exit if it returns `false`. 
+
+##### Logging Client
+
+The `Logging` client is now accessible from the `service.LoggingClient()` API. 
+
+!!! note "New extended Logging Client API"
+    The Logging Client API now has `formated` versions of all the logging APIs, which are `Infof`, `Debugf`, `Tracef`, `Warnf `and `Errorf`. If your code uses `fmt.Sprintf` to format your log messages then it can now be simplified by using these new APIs.
+
+##### Application Settings
+
+The access functions for retrieving the service's custom Application Settings (`ApplicationSettings`, `GetAppSettingStrings`,  and `GetAppSetting` ) have not changed. An improved capability to have structured custom configuration has been added. See the [Structure Custom Configuration](../AdvancedTopics/#structure-custom-configuration) section for more details.
+
+##### Functions Pipeline
+
+Setting the  Functions Pipeline has not changed, but the name of some built in functions have changed and new ones have been added. See the [Built-In Pipeline Functions](../BuiltIn) section for more details.
+
+!!! example "Example - Setting Functions Pipeline"
+
+    ```go
+    if err := service.SetFunctionsPipeline(
+    	transforms.NewFilterFor(deviceNames).FilterByDeviceName,
+    	transforms.NewConversion().TransformToXML,
+    	transforms.NewHTTPSender(exportUrl, "application/xml", false).HTTPPost,
+    ); err != nil {
+    	lc.Errorf("SetFunctionsPipeline returned error: %s", err.Error())
+    	os.Exit(-1)
+    }
+    ```
+
+##### MakeItRun
+
+The `MakeItRun` API has not changed.
+
+!!! example "Example - Call to MakeItRun"
+
+    ```go
+    err = service.MakeItRun()
+    if err != nil {
+    	lc.Errorf("MakeItRun returned error: %s", err.Error())
+    	os.Exit(-1)
+    }
+    ```
+
+#### Custom Pipeline Functions
+
+##### Pipeline Function signature
+
+The major change to custom Pipeline Functions for EdgeX 2.0 is the new function signature which drives all the other changes.
+
+!!! example "Example - New Pipeline Function signature"
+
+    ```go
+    type AppFunction = func(cxt AppFunctionContext, data interface{}) (bool, interface{})
+    ```
+
+This function signature passes in an instance of the new AppFunctionContext API for the context and now has only a single `data` instance for the function to operate on.
+
+##### Return Values
+
+The Pipeline Function return values definitions have not changed.
```

Comment:
I don't think you should have two successive plural nouns here

Reply:
Yea, I struggled with that. There are multiple return values... I will reword to avoid this.

Reply:
`The definitions for the Pipeline Function return values have not changed.`